### PR TITLE
Tolerating RTTI in OMR

### PR DESCRIPTION
**Pull request by zheddie**
Adding --enable-OMR_RTTI option to enable OMR tolearting the RTTI in case language specific VM require RTTI feature .

Issue: #646
Signed-off-by: Gavin Zhang <zheddie@163.com>

### CODE REVIEW

File: omrmakefiles/rules.linux.mk [hunk 1]
```diff
@@ -57,7 +57,11 @@ endif
 
 # Compile without exceptions
 ifeq (gcc,$(OMR_TOOLCHAIN))
-    GLOBAL_CXXFLAGS+=-fno-exceptions -fno-rtti -fno-threadsafe-statics
+ifeq (1,$(OMR_RTTI))
+    GLOBAL_CXXFLAGS+=-fno-exceptions -fno-threadsafe-statics
```
Comment (charliegracie) [inline omrmakefiles/rules.linux.mk]: There appears to be inconsistent spacing/tabbing between line 61 and 63.  Would you be able to update this?

File: omrmakefiles/rules.linux.mk [hunk 2]
```diff
@@ -57,7 +57,11 @@ endif
...
+else
+	GLOBAL_CXXFLAGS+=-fno-exceptions -fno-rtti -fno-threadsafe-statics
+endif
 else
     GLOBAL_CXXFLAGS+=-qnortti -qnoeh
```
Comment (charliegracie) [inline omrmakefiles/rules.linux.mk]: -qnortti is also a flag for controlling RTTI when compiling with xlc.  Would you be able to make a similar change for this?

**Comment by charliegracie** [comment]
I did a quick scan of the codebase tonight to see if there were any other rules.mk files setting compiler specific options for turning of RTTI but I did not see any. I will give it another look tomorrow to make sure this PR catches all of them.

**Comment by zheddie** [comment]
@charliegracie  Refined basing on your comments. thx.

**Comment by charliegracie** [comment]
Hi @zheddie would you be able to squash the 5 commits into a single commit?  Once you do this you will have to force push to your GitHub branch. Thanks

**Comment by zheddie** [comment]
Thx @charliegracie , update was done based on your comments . Please review it.thx.

**Pull request merged**